kivy/uix/textinput.py

```diff
@@ -2850,12 +2842,19 @@ def keyboard_on_key_up(self, window, keycode):
         key = keycode[0]
         k = self.interesting_keys.get(key)
         if k:
-            key = (None, None, k, 1)
-            self._key_up(key)
+            self._key_up(k)
 
     def keyboard_on_textinput(self, window, text):
         if self._selection:
             self.delete_selection()
+        if self._ime_composition:
+            self._replace_ime(text)
+            self._ime_composition = ''
+        if text.endswith('\n') and not self.multiline:
```

Comment:
Do we handle `\r\n`?

Reply:
I don’t know if we need to, this happens on android when you press enter after swiping a word, the word is sent again, but with \n at the end, but maybe we can add it to be safe.

Reply:
Ok, either works fine I guess (windows doesn't have swipe I imagine).